common/dynamicconfig/constants.go

```diff
@@ -519,9 +519,15 @@ const (
 	HistoryPersistenceDynamicRateLimitingParams = "history.persistenceDynamicRateLimitingParams"
 	// HistoryLongPollExpirationInterval is the long poll expiration interval in the history service
 	HistoryLongPollExpirationInterval = "history.longPollExpirationInterval"
+	// HistoryCacheSizeBasedLimit if true, HistoryCacheMaxSize and HistoryCacheHostLevelMaxSize are the total size of
+	// the objects in history cache. Otherwise, HistoryCacheMaxSize and HistoryCacheHostLevelMaxSize are the number of
+	// entries in the history cache.
+	HistoryCacheSizeBasedLimit = "history.cacheSizeBasedLimit"
 	// HistoryCacheInitialSize is initial size of history cache
 	HistoryCacheInitialSize = "history.cacheInitialSize"
-	// HistoryCacheMaxSize is max size of history cache
+	// HistoryCacheMaxSize is max size of history cache. If history.cacheSizeBasedLimit set to true, this value is the
+	// maximum size of history cache in bytes. if history.cacheSizeBasedLimit set to false, this value is the maximum number of
+	// entries in the cache.
```

Comment:
nit: I don't have a strong opinion, though I feel separate configurations would make operation easier.

Reply:
I agree. I'm not a fan of config options whose interpretations change based on the value of another config

Reply:
Yeah, that makes sense.